### ISSUE TITLE
loopProtect is not defined problem on beta

### ISSUE THREAD

**Issue by QuincyLarson** [freeCodeCamp/freeCodeCamp#16260]
With some of the challenges, such as the algorithm challenges, we're experiencing this problem:

![learn_to_code___freecodecamp](https://user-images.githubusercontent.com/985197/34286676-c09239a6-e6a7-11e7-80d3-4c8b32eaf09d.png)

This may be related to https://github.com/freeCodeCamp/freeCodeCamp/issues/13532, but I'm experiencing this in edge Chrome (Version 63.0.3239.84 (Official Build) (64-bit) on MacOS)

**Comment by scissorsneedfoodtoo** [comment]
I was poking around the developer console in Chrome on both freecodecamp.org and beta.freecodecamp.org. For some reason it seems like the following code is missing from the <iframe>#document<html><head><script> tag on the beta site:

  window.loopProtect = parent.loopProtect;
  window.__err = null;
  window.loopProtect.hit = function(line) {
    window.__err = new Error(
      'Potential infinite loop at line ' +
      line +
      '. To disable loop protection, write:' +
      ' \n\/\/ noprotect\nas the first' +
      ' line. Beware that if you do have an infinite loop in your code' +
      ' this will crash your browser.'
    );
  };


This code is present on freecodecamp.org. When I add it to the editor on the beta site I no longer get the "loopProtect is not defined" error.

![image](https://user-images.githubusercontent.com/2051070/34288344-18b31a5a-e730-11e7-93a1-130aa8850044.png)

**Comment by scissorsneedfoodtoo** [comment]
I think I have a fix, though I'm really unsure if it's best practice or breaks anything.

Originally I tried editing common/app/routes/Challenges/utils/frame.js and adding the loopProtect code wrapped in a `<script>` tag to the header there. My thinking was to try to replicate the structure of the www.freecodecamp.org challenge pages. However, that didn't work (maybe it's out of scope?).

After trying some other things, I ended up changing the app/routes/Challenges/rechallenge/builders.js file, and putting the loopProtect code above the default content that gets rendered to the code editor on the page. The end result is that loopProtect is within the same script tag as the default challenge content, i.e. `function sumAll(arr) {...`, but is not visible on the page. Doing this seems to solve the loopProtect undefined error, but I'll need to do some further testing.
